### PR TITLE
[milight] Prevent V3 bridge getting stuck in the offline state.

### LINKED ISSUE #4609: [milight] V3 bridges remain offline following router reboot
Following the upgrade to 2.4, I've noticed that my milight V3 bridges go offline (communication error) and stay offline if I reboot my router.

To get them going again I need to disable/enable the bridge thing or restart OH.

This issue did not occur on 2.2.

### PR DESCRIPTION
**Pull request by MikeJMajor**
Fixes #4609

An issue introduced in 2.4 meant that a V3 bridge could go offline and
remain offline following a router reboot.

Also tweaked the keep-alive logic to use 5 attempts with the delay between attempts ranging from 500ms to 2500ms. In my environment this keeps my bridges permanently online and resolves the 5-10 spurious offline/online blips I would see throughout the day.

Signed-off-by: Mike Major <mike_j_major@hotmail.com>

### CODE REVIEW

**Comment by MikeJMajor** [comment]
@davidgraeff could you please take a look at this.

**Comment by davidgraeff** [comment]
This PR basically changes back my idea of relieving the openHAB scheduler by using a socket timeout.
Hm. Can we really not avoid using the scheduler?

**Comment by MikeJMajor** [comment]
Is there a particular reason why we shouldn't be using the scheduler in this scenario? It's not an important thread and doesn't really matter if it's delayed. I'm sure I've spotted in other binding reviews that it was preferred for a binding to use the scheduler rather than creating it's own threads.

Given the socket timeout won't work when the network is unreachable it seemed messy to me to be using a combination of socket timeout and sleep to implement the delay. 

Your original change mentioned using the socket timeout to make the code simpler. Using the scheduler has kept changes to your code to a minimum and seemed a lot simpler than adding in a thread sleep and having to worry about stopping the thread cleanly. I would have needed to have a running flag per thread since I could see a scenario when it could be set back to true again before the previous thread had terminated.

File: addons/binding/org.openhab.binding.milight/src/main/java/org/openhab/binding/milight/internal/handler/BridgeV3Handler.java [hunk 1]
```diff
@@ -85,12 +88,13 @@ protected void startConnectAndKeepAlive() {
...
 
-        new Thread(this::receive).start();
-        running = true;
+        running = scheduler.scheduleWithFixedDelay(this::receive, 0, config.refreshTime, TimeUnit.SECONDS);
     }
 
     protected void stopKeepAlive() {
-        running = false;
+        if (running != null) {
+            running.cancel(false);
```
Comment (davidgraeff) [inline addons/binding/org.openhab.binding.milight/src/main/java/org/openhab/binding/milight/internal/handler/BridgeV3Handler.java]: Could you also set `running` to `null` here, to allow the garbage collector to collect this object?

Comment (MikeJMajor) [reply addons/binding/org.openhab.binding.milight/src/main/java/org/openhab/binding/milight/internal/handler/BridgeV3Handler.java]: Done

**Comment by davidgraeff** [comment]
Thanks :)

**Pull request merged**